jmclient/jmclient/wallet_service.py

```diff
@@ -225,8 +225,7 @@ def transaction_monitor(self):
             # transaction pertains to anything known (but must
             # have correct label per above); filter on this Joinmarket wallet label,
             # or the external monitoring label:
-            if "label" in tx and tx["label"] in [
-                self.EXTERNAL_WALLET_LABEL, self.get_wallet_name()]:
+            if self.bci.is_address_labeled(tx, self.get_wallet_name()):
```

Comment:
You need to check whether the transaction has the EXTERNAL_WALLET_LABEL also; have I missed something or did you drop this?

Reply:
My mistake, fixed.